azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -121,16 +143,94 @@ protected void deployArtifacts() throws AzureAuthFailureException, InterruptedEx
             } else {
                 target = new WebAppDeployTarget(app);
             }
-            getFactory().getArtifactHandler(this).publish(target);
+            final ArtifactHandler artifactHandler = getFactory().getArtifactHandler(this);
+            final boolean isV1Schema = SchemaVersion.fromString(this.getSchemaVersion()) == SchemaVersion.V1;
+            if (isV1Schema) {
+                handleV1Resources(artifactHandler);
+            } else {
+                final List<Resource> v2Resources = this.deployment == null ? null : this.deployment.getResources();
+
+                if (v2Resources == null || v2Resources.isEmpty()) {
```

Comment:
why don't we check the resources in handle Function like V1?

Reply:
Because if (v2Resources == null || v2Resources.isEmpty()) the method deployArtifacts will return, in method handleV2Resources, there is no way to return for deployArtifacts